conda/cli/main.py

```diff
@@ -103,6 +104,14 @@ def main(*args, **kwargs):
             if argv1.startswith('shell.'):
                 from ..activate import main as activator_main
                 return activator_main()
+            elif argv1.startswith('shell_support.'):
+                # Hidden commands to provide metadata to shells.
+                if argv1 == 'shell_support.commands':
+                    for command in sorted(
+                            find_builtin_commands() + list(find_commands(True))
+                    ):
+                        print(command)
+                    return
```

Comment:
I'd rather implement this in `activate.py`.  Add a new subcommand there, by adding a method to `_Activator`.  See what we did with `hook` for example.  I'm fine just adding a `commands` subcommand there for now.  So instead of `conda shell_support.commands` you'd use `conda shell.ANYOFTHEM commands`.  I consider the original `shell.` "hidden" and really there to provide support for shells :)

The "for now" part of my comment was that we might want to think through more generally and thoroughly what `shell_support.commands` is getting at, namely helpers for tab completion.  For example, how we want this to work after a user types `conda env[TAB][TAB]` (assuming `conda env` was actually hooked into the main parser, which it will be someday).  The general case might just be build out a `conda shell.bash completions "$@"` option.

For reference, besides what's in the PR, the most advanced completions we have right now are what's in `conda.fish`.  We have none for bash.  However, one thing `conda init` will give us, via use of `conda shell.XXX hook`, is inject those completions as they're developed and added.